waku/protocol/v2/waku_store.nim

```diff
@@ -90,3 +96,17 @@ proc subscription*(proto: WakuStore): MessageNotificationSubscription =
     proto.messages.add(msg)
 
   MessageNotificationSubscription.init(@[], handle)
+
+proc query*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc) {.async, gcsafe.} =
+  let conn = await w.switch.dial(w.peerInfo.peerId, w.peerInfo.addrs, WakuStoreCodec)
+
+  await conn.writeLP(query.encode().buffer)
+
+  var message = await conn.readLp(64*1024)
```

Comment:
I don't understand where the this message is generate. How are messages stored in memory and where does  initial HistoryResponse get generated on server side?

Reply:
@oskarth its created here: https://github.com/status-im/nim-waku/pull/174/files#diff-86781df6cf9d09f0ac3ab22aea69a517R59

Reply:
ah duh, right